pkg/client/apiutil/errors.go

```diff
@@ -52,3 +52,9 @@ func (e *ErrResourceDiscoveryFailed) Unwrap() []error {
 	}
 	return subErrors
 }
+
+// Is makes it possible for the callers to use `errors.Is(` helper on errors wrapped with ErrResourceDiscoveryFailed error.
+func (e *ErrResourceDiscoveryFailed) Is(target error) bool {
+	_, ok := target.(*ErrResourceDiscoveryFailed)
+	return ok
```

Comment:
```suggestion
       ep := &ErrResourceDiscoveryFailed{}
       return errors.As(target,ep)
```

We have done this for Terminal error in `pkg/reconcile/reconcile.go:179`
https://github.com/kubernetes-sigs/controller-runtime/blob/e6c3d139d2b6c286b1dbba6b6a95919159cfe655/pkg/reconcile/reconcile.go#L179-L182

I am not sure if this is needed but I would add a test if you are adding this change.

Reply:
<img width="993" alt="Screenshot 2024-08-29 at 1 06 35 PM" src="https://github.com/user-attachments/assets/0a97f4b3-89de-4680-b4ce-179f43cb6fc6">

I don't think `Is` should be implemented using `As`, since `As` calls unwrap.

Reply:
It doesn't matter, if the `Is` implementation doesn't call unwrap, `errors.Is` does, so the result is the same. Please add a test though.